### PR TITLE
fix commands to work from WP 5.7 to 6.4

### LINKED ISSUE #118: WP 6.5 breaks `openDocumentSettingsSidebar` command
### Describe the bug

`openDocumentSettingsSidebar` is not longer working because the markup of the sidebar component has changed. Most likely as a result of https://github.com/WordPress/gutenberg/commit/c90bb031ee5ee72ec382c10a123cd713e9c78f65#diff-790eee3f502d6ff6b5b298913a7ea881c6e556c969bda2507f0e4c915005ad37L49-R37

### Steps to Reproduce

1. Checkout WP 6.5
2. Create a test and utilize the `openDocumentSettingsSidebar` command
3. Observe that test fails

### Screenshots, screen recording, code snippet

image

### Environment information

_No response_

### WordPress information

`
### wp-core ###

version: 6.5-alpha-57541
site_language: en_US
user_language: en_US
timezone: +00:00
permalink: undefined
https_status: false
multisite: false
user_registration: 0
blog_public: 1
default_comment_status: open
environment_type: local
user_count: 1
dotorg_communication: true

### wp-paths-sizes ###

wordpress_path: /var/www/html
wordpress_size: 225.49 MB (236448033 bytes)
uploads_path: /var/www/html/wp-content/uploads
uploads_size: 2.16 MB (2268718 bytes)
themes_path: /var/www/html/wp-content/themes
themes_size: 36.86 MB (38645503 bytes)
plugins_path: /var/www/html/wp-content/plugins
plugins_size: 258.25 MB (270795522 bytes)
database_size: 2.13 MB (2228224 bytes)
total_size: 524.89 MB (550386000 bytes)

### wp-active-theme ###

name: Twenty Twenty-Four (twentytwentyfour)
version: 1.0
author: the WordPress team
author_website: https://wordpress.org
parent_theme: none
theme_features: core-block-patterns, post-thumbnails, responsive-embeds, editor-styles, html5, automatic-feed-links, widgets-block-editor, block-templates
theme_path: /var/www/html/wp-content/themes/twentytwentyfour
auto_update: Disabled

### wp-themes-inactive (13) ###

Twenty Eleven: version: 4.5, author: the WordPress team, Auto-updates disabled
Twenty Fifteen: version: 3.6, author: the WordPress team, Auto-updates disabled
Twenty Fourteen: version: 3.8, author: the WordPress team, Auto-updates disabled
Twenty Nineteen: version: 2.7, author: the WordPress team, Auto-updates disabled
Twenty Seventeen: version: 3.5, author: the WordPress team, Auto-updates disabled
Twenty Sixteen: version: 3.1, author: the WordPress team, Auto-updates disabled
Twenty Ten: version: 4.0, author: the WordPress team, Auto-updates disabled
Twenty Thirteen: version: 4.0, author: the WordPress team, Auto-updates disabled
Twenty Twelve: version: 4.1, author: the WordPress team, Auto-updates disabled
Twenty Twenty: version: 2.5, author: the WordPress team, Auto-updates disabled
Twenty Twenty-One: version: 2.1, author: the WordPress team, Auto-updates disabled
Twenty Twenty-Three: version: 1.3, author: the WordPress team, Auto-updates disabled
Twenty Twenty-Two: version: 1.6, author: the WordPress team, Auto-updates disabled

### wp-plugins-active (1) ###

Curated Query Loop: version: 0.1.0, author: Peter Sorensen, 10up, Auto-updates disabled

### wp-plugins-inactive (1) ###

Hello Dolly: version: 1.7.2, author: Matt Mullenweg, Auto-updates disabled

### wp-media ###

image_editor: WP_Image_Editor_Imagick
imagick_module_version: 1691
imagemagick_version: ImageMagick 6.9.11-60 Q16 aarch64 2021-01-25 https://imagemagick.org
imagick_version: 3.6.0
file_uploads: 1
post_max_size: 1G
upload_max_filesize: 1G
max_effective_size: 1 GB
max_file_uploads: 20
imagick_limits: 
	imagick::RESOURCETYPE_AREA: 122 MB
	imagick::RESOURCETYPE_DISK: 1073741824
	imagick::RESOURCETYPE_FILE: 786432
	imagick::RESOURCETYPE_MAP: 512 MB
	imagick::RESOURCETYPE_MEMORY: 256 MB
	imagick::RESOURCETYPE_THREAD: 1
	imagick::RESOURCETYPE_TIME: 9.2233720368548E+18
imagemagick_file_formats: 3FR, 3G2, 3GP, AAI, AI, APNG, ART, ARW, AVI, AVIF, AVS, BGR, BGRA, BGRO, BIE, BMP, BMP2, BMP3, BRF, CAL, CALS, CANVAS, CAPTION, CIN, CIP, CLIP, CMYK, CMYKA, CR2, CR3, CRW, CUR, CUT, DATA, DCM, DCR, DCX, DDS, DFONT, DNG, DPX, DXT1, DXT5, EPDF, EPI, EPS, EPS2, EPS3, EPSF, EPSI, EPT, EPT2, EPT3, ERF, FAX, FILE, FITS, FRACTAL, FTP, FTS, G3, G4, GIF, GIF87, GRADIENT, GRAY, GRAYA, GROUP4, H, HALD, HDR, HEIC, HISTOGRAM, HRZ, HTM, HTML, HTTP, HTTPS, ICB, ICO, ICON, IIQ, INFO, INLINE, IPL, ISOBRL, ISOBRL6, J2C, J2K, JBG, JBIG, JNG, JNX, JP2, JPC, JPE, JPEG, JPG, JPM, JPS, JPT, JSON, K25, KDC, LABEL, M2V, M4V, MAC, MAGICK, MAP, MASK, MAT, MATTE, MEF, MIFF, MKV, MNG, MONO, MOV, MP4, MPC, MPG, MRW, MSL, MTV, MVG, NEF, NRW, NULL, ORF, OTB, OTF, PAL, PALM, PAM, PATTERN, PBM, PCD, PCDS, PCL, PCT, PCX, PDB, PDF, PDFA, PEF, PES, PFA, PFB, PFM, PGM, PGX, PICON, PICT, PIX, PJPEG, PLASMA, PNG, PNG00, PNG24, PNG32, PNG48, PNG64, PNG8, PNM, POCKETMOD, PPM, PREVIEW, PS, PS2, PS3, PSB, PSD, PTIF, PWP, RADIAL-GRADIENT, RAF, RAS, RAW, RGB, RGBA, RGBO, RGF, RLA, RLE, RMF, RW2, SCR, SCT, SFW, SGI, SHTML, SIX, SIXEL, SPARSE-COLOR, SR2, SRF, STEGANO, SUN, TEXT, TGA, THUMBNAIL, TIFF, TIFF64, TILE, TIM, TTC, TTF, TXT, UBRL, UBRL6, UIL, UYVY, VDA, VICAR, VID, VIDEO, VIFF, VIPS, VST, WBMP, WEBM, WEBP, WMV, WPG, X, X3F, XBM, XC, XCF, XPM, XPS, XV, XWD, YCbCr, YCbCrA, YUV
gd_version: bundled (2.1.0 compatible)
gd_formats: GIF, JPEG, PNG, WebP, BMP
ghostscript_version: 10.00.0

### wp-server ###

server_architecture: Linux 5.15.49-linuxkit aarch64
httpd_software: Apache/2.4.57 (Debian)
php_version: 8.2.15 64bit
php_sapi: apache2handler
max_input_variables: 1000
time_limit: 30
memory_limit: 128M
admin_memory_limit: 256M
max_input_time: -1
upload_max_filesize: 1G
php_post_max_size: 1G
curl_version: 7.88.1 OpenSSL/3.0.11
suhosin: false
imagick_availability: true
pretty_permalinks: true
htaccess_extra_rules: false
current: 2024-02-10T21:27:49+00:00
utc-time: Saturday, 10-Feb-24 21:27:49 UTC
server-time: 2024-02-10T21:27:48+00:00

### wp-database ###

extension: mysqli
server_version: 11.2.2-MariaDB-1:11.2.2+maria~ubu2204
client_version: mysqlnd 8.2.15
max_allowed_packet: 16777216
max_connections: 151

### wp-constants ###

WP_HOME: http://localhost:8888
WP_SITEURL: http://localhost:8888
WP_CONTENT_DIR: /var/www/html/wp-content
WP_PLUGIN_DIR: /var/www/html/wp-content/plugins
WP_MEMORY_LIMIT: 40M
WP_MAX_MEMORY_LIMIT: 256M
WP_DEBUG: true
WP_DEBUG_DISPLAY: true
WP_DEBUG_LOG: false
SCRIPT_DEBUG: true
WP_CACHE: false
CONCATENATE_SCRIPTS: undefined
COMPRESS_SCRIPTS: undefined
COMPRESS_CSS: undefined
WP_ENVIRONMENT_TYPE: local
WP_DEVELOPMENT_MODE: undefined
DB_CHARSET: utf8
DB_COLLATE: undefined

### wp-filesystem ###

wordpress: writable
wp-content: writable
uploads: writable
plugins: writable
themes: writable

`

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Sidsector9**
### Description of the Change

This PR:
- fixes the `insertBlock`, `createPost` and `openDocumentSettingSidebars` commands.
- updates NPM dependencies.
- Renames `check-block-pattern-exists.test.js` to `z.check-block-pattern-exists.test.js` as the order of this test affects the outcome of other tests. (I couldn't figure out the reason yet.)

### How to test the Change

Confirm in the E2E log if this command passes in the 3 WP versions.

### Changelog Entry

> Fixed - Failing `insertBlock` command

### Credits

Props @Sidsector9 

### Checklist:

- [ ] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "prettify": "prettier --write \"**/*.{ts,js}\"",
     "typecheck": "tsc --noEmit",
     "cypress:open": "cypress open --config-file tests/cypress/cypress-config.js --e2e --browser chrome",
-    "cypress:run": "cypress run --config-file tests/cypress/cypress-config.js",
+    "cypress:run": "cypress run --config-file tests/cypress/cypress-config.js --browser chrome",
```
Comment (iamdharmesh) [inline package.json]:
```suggestion
    "cypress:run": "cypress run --config-file tests/cypress/cypress-config.js",
```

Could we keep running this as before? as most of our plugins don't explicitly run tests in chrome?

**Review by iamdharmesh — COMMENTED** [review]
@Sidsector9 Tests are passing on the GH action and PR LGTM. just added one minor comment to discuss. once it is concluded we are good to merge this.

Thanks for fixing this.

**Comment by jeffpaul** [comment]
@Sidsector9 @iamdharmesh is the `WP latest` failure a concern here or good to merge this in?

**Comment by peterwilsoncc** [comment]
The WP Latest test failures are intermittent both locally and in the GH Actions. Running in the browser I am seeing some issues with the rendering of the editors iframe but I am unsure if that is happening in the GH actions too.

In the most recent push, the [first run](https://github.com/10up/cypress-wp-utils/actions/runs/7880789040/job/21503379082?pr=117#step:7:168) fails with `Create Post > Should be able to create Page`

In the [second run](https://github.com/10up/cypress-wp-utils/actions/runs/7880789040/job/21504449088#step:7:201) both `Create Post > Should be able to create Page` and `Create Post > Should be able to create Draft Page` fail.

Similar happening elsewhere.

**Comment by darylldoyle** [comment]
@peterwilsoncc I've taken a look at this and whilst I can't replicate the error locally, I do see the following in the artifacts from the latest GH Actions failure:

Command checkPostExists -- Post 9d0f6af9 should exist -- before all hook (failed)

This seems to me that something is causing the editor to fall over. Is there anyway to get more information about general JS errors rather than Cypress specific errors? That will likely give us more insight into what's going on under the hood.

**Comment by peterwilsoncc** [comment]
I've been able to reproduce locally using the electron browser rather than Chrome but not seeing any hints in the console when running `cypress open`.

**Comment by darylldoyle** [comment]
Thanks @peterwilsoncc! Running in Electron I was able to reproduce that error. If I enable `SCRIPT_DEBUG`, I can see the following error.

image


This error points at this line: 

https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/image/use-client-width.js#L20

Interestingly, if you log out `ref.current.ownerDocument`, the `defaultView` property is null:

Screenshot 2024-02-15 at 14 31 23

Yet if you log out the extracted `defaultView` const, you get this:

image

but notice how `removeEventListener()` isn't a function defined on that object 🤔 

I'm assuming this is a bug in Electron browser windows, but I'm not sure where that leaves us or this test.

**Comment by peterwilsoncc** [comment]
@darylldoyle Thank you! That's a great help.

In 9fffc473630c5be441717cb10e53d10787e93993 I've switched the GitHub actions to use Chrome and will run the test suite a few times to see if it ends up less flaky.

It's probably a good thing to do as it better represents what the end user is running and that's the purpose of the test suite.

**Review by peterwilsoncc — DISMISSED** [review]
I think this is good to go now the flakey tests have been resolved by using chrome on GitHub.

It could probably do with another approval as I've pushed a few changes:

* updated winamp plugin
* modified plugin test to allow for versions of WP unsupported by the plugin

**Comment by Sidsector9** [comment]
cc: @iamdharmesh we've switched back to Chrome 😄

**Review by iamdharmesh — APPROVED** [review]

**Pull request merged**